### PR TITLE
Clarify nullable types for constants

### LINKED ISSUE #446: Why is a constant type nullable?
https://heycam.github.io/webidl/#prod-ConstType

```
ConstType ::
    PrimitiveType Null
    identifier Null
```

https://heycam.github.io/webidl/#idl-nullable-type

```webidl
interface MyConstants {
  const boolean? ARE_WE_THERE_YET = false;
};
```

I couldn't find a real world example that uses nullable constant types. Why is it nullable? How is `const boolean?` different from `const boolean`?

### PR DESCRIPTION
**Pull request by TimothyGu**
Fixes #446.

/cc @SaschaNaz

***
[Preview](https://s3.amazonaws.com/pr-preview/TimothyGu/webidl/nullable-constant.html) | [Diff](https://s3.amazonaws.com/pr-preview/heycam/webidl/f97da39...TimothyGu:3c4eb7b.html)

### CODE REVIEW

File: index.bs [hunk 1]
```diff
@@ -5508,8 +5513,8 @@ the string “OrNull”.
     is written as <code class="idl">boolean?</code>:
 
     <pre highlight="webidl">
-        interface MyConstants {
-          const boolean? ARE_WE_THERE_YET = false;
+        interface NetworkFetcher {
+          void get(optional boolean? areWeThereYet = false);
```
Comment (tobie) [inline index.bs]: Why is the constant gone from this example?

File: index.bs [hunk 2]
```diff
@@ -1167,6 +1167,11 @@ If an [=identifier=] is used,
 it must reference a [=typedef=]
 whose type is a primitive type or a nullable primitive type.
 
+Note: For null to be able to become the value of a constant, at least one [=nullable type=] needs to
```
Comment (tobie) [inline index.bs]: I'm unconvinced this really clarifies, tbh.

File: index.bs (hunk 1 above)
Comment (TimothyGu) [reply index.bs]: This example is used to illustrate how nullable types work, not how constants work. Plus, https://github.com/heycam/webidl/issues/446#issuecomment-331388615.

Comment (tobie) [reply index.bs]: Oops. Sorry. I didn't know we had const scattered across examples in the spec. :-/

**Pull request merged**